### ISSUE TITLE
boost-python3: build failed

### ISSUE THREAD

**Issue by Char-Aznable** [Linuxbrew/homebrew-core#10289]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single formula? If it's a general `brew` problem please file this issue at Linuxbrew/brew: https://github.com/Linuxbrew/brew/issues/new/choose. If it's a tap (e.g. Brewsci/homebrew-bio) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
brew upgrade boost-python3
- What happened (include command output)
Build failed with:
```
Last 15 lines from /home/aznb/.cache/boost-python3/01.python3:                   
 #define PyDataMem_FREE(ptr)  free(ptr)                                                                                                                        
                              ^                                                                                                                                
numpy/core/src/multiarray/alloc.c:241:1: note: in expansion of macro 'PyDataMem_FREE'                                                                          
 PyDataMem_FREE(void *ptr)                                                                                                                                     
 ^                                                                                                                                                             
Running from numpy source directory.                                                                                                                           
/tmp/56460047.1.long.q/boost-python3--numpy-20181110-168402-usla8m/numpy-1.14.5/numpy/distutils/system_info.py:624: UserWarning:                               
    Atlas (http://math-atlas.sourceforge.net/) libraries not found.                                                                                            
    Directories to search for the libraries can be specified in the                                                                                            
    numpy/distutils/site.cfg file (section [atlas]) or by setting                                                                                              
    the ATLAS environment variable.                                                                                                                            
  self.calc_info()                                                                                                                                             
/home/aznb/.linuxbrew/opt/python/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'define_macros'             
  warnings.warn(msg)                                                                                                                                           
error: Command "gcc-5 -Wno-unused-result -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -DNPY_INTERNAL_BUILD=1 -DHAVE_NPY_CONFIG_H=1 -D_FILE_OFFSET_BITS=64 -D_LARG
EFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -DNO_ATLAS_INFO=1 -DHAVE_CBLAS -I/home/aznb/.linuxbrew/opt/openblas/lib -I/usr/local/include -I/usr/i
nclude -I/usr/include/suitesparse -I/home/aznb/.linuxbrew/opt/python/include -Ibuild/src.linux-x86_64-3.7/numpy/core/src/private -Inumpy/cor
e/include -Ibuild/src.linux-x86_64-3.7/numpy/core/include/numpy -Inumpy/core/src/private -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/sr
c/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -I/home/aznb/.linuxbrew/opt/
openssl/include -I/home/aznb/.linuxbrew/opt/python/include/python3.7m -Ibuild/
src.linux-x86_64-3.7/numpy/core/src/private -Ibuild/src.linux-x86_64-3.7/numpy/core/src/npymath -Ibuild/src.linux-x86_64-3.7/numpy/core/src/private -Ibuild/src
.linux-x86_64-3.7/numpy/core/src/npymath -Ibuild/src.linux-x86_64-3.7/numpy/core/src/private -Ibuild/src.linux-x86_64-3.7/numpy/core/src/npymath -c numpy/core/
src/multiarray/alloc.c -o build/temp.linux-x86_64-3.7/numpy/core/src/multiarray/alloc.o -MMD -MF build/temp.linux-x86_64-3.7/numpy/core/src/multiarray/alloc.o.
d" failed with exit status 1
```
Examining the 01.python3 reveals numpy was picking up ancient version of blas from the OS.  Another problem with the formula is relapsing https://github.com/Linuxbrew/homebrew-core/pull/6095

The following patch can fix the build
```
diff --git a/Formula/boost-python3.rb b/Formula/boost-python3.rb
index 9ac1860edf..0a63442a05 100644
--- a/Formula/boost-python3.rb
+++ b/Formula/boost-python3.rb
@@ -45,18 +45,29 @@ class BoostPython3 < Formula
     inreplace "bootstrap.sh", "using python", "#using python"
 
     pyver = Language::Python.major_minor_version "python3"
-    py_prefix = Formula["python3"].opt_frameworks/"Python.framework/Versions/#{pyver}"
+    py_prefix = Formula["python3"].opt_prefix
 
     numpy_site_packages = buildpath/"homebrew-numpy/lib/python#{pyver}/site-packages"
     numpy_site_packages.mkpath
+
     ENV["PYTHONPATH"] = numpy_site_packages
     resource("numpy").stage do
+      openblas = Formula["openblas"].opt_prefix
+      ENV["ATLAS"] = "None" # avoid linking against Accelerate.framework
+      ENV["BLAS"] = ENV["LAPACK"] = "#{openblas}/lib/libopenblas.so"
+      config = <<~EOS
+        [openblas]
+        libraries = openblas
+        library_dirs = #{openblas}/lib
+        include_dirs = #{openblas}/include
+      EOS
+      Pathname("site.cfg").write config
       system "python3", *Language::Python.setup_install_args(buildpath/"homebrew-numpy")
```

I take it that no one takes care of merging formula properly from homebrew to linuxbrew? This kind of relapsing issues is very common to linuxbrew.

**Comment by iMichka** [comment]
I am doing the merges during the week. It takes me around 8 to 10 hours per week on my free time. I try to break as less as possible, but mistakes do happen sometimes. Pull requests are welcome to get this fixed.

**Comment by Char-Aznable** [comment]
Didn't mean to complain. I was just trying to understand why the issue happened. It would be nice to have  the formula tested maybe by the CI bot before releasing to the users. Things like this makes it practically impossible to run brew upgrade. I had to  navigate through 10 other broken formula like this to pin down the issue...

**Comment by iMichka** [comment]
Fixed by #10390
